bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/VoiceManager.java

```diff
@@ -0,0 +1,584 @@
+                    getSink(), localeProvider.getLocale(), keyword);
+            processor.start();
+        } else {
+            String msg = "Cannot start dialog as services are missing.";
+            logger.error(msg);
+            throw new IllegalStateException(msg);
+        }
+    }
+
+    protected void setLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = localeProvider;
+    }
+
+    protected void unsetLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = null;
+    }
+
+    protected void addKSService(KSService ksService) {
+        this.ksServices.put(ksService.getId(), ksService);
+    }
+
+    protected void removeKSService(KSService ksService) {
+        this.ksServices.remove(ksService.getId());
+    }
+
+    protected void addSTTService(STTService sttService) {
+        this.sttServices.put(sttService.getId(), sttService);
+    }
+
+    protected void removeSTTService(STTService sttService) {
+        this.sttServices.remove(sttService.getId());
+    }
+
+    protected void addTTSService(TTSService ttsService) {
+        this.ttsServices.put(ttsService.getId(), ttsService);
+    }
+
+    protected void removeTTSService(TTSService ttsService) {
+        this.ttsServices.remove(ttsService.getId());
+    }
+
+    protected void addHumanLanguageInterpreter(HumanLanguageInterpreter humanLanguageInterpreter) {
+        this.humanLanguageInterpreters.put(humanLanguageInterpreter.getId(), humanLanguageInterpreter);
+    }
+
+    protected void removeHumanLanguageInterpreter(HumanLanguageInterpreter humanLanguageInterpreter) {
+        this.humanLanguageInterpreters.remove(humanLanguageInterpreter.getId());
+    }
+
+    protected void addAudioSource(AudioSource audioSource) {
+        this.audioSources.put(audioSource.getId(), audioSource);
+    }
+
+    protected void removeAudioSource(AudioSource audioSource) {
+        this.audioSources.remove(audioSource.getId());
+    }
+
+    protected void addAudioSink(AudioSink audioSink) {
+        this.audioSinks.put(audioSink.toString(), audioSink);
+    }
+
+    protected void removeAudioSink(AudioSink audioSink) {
+        this.audioSinks.remove(audioSink.toString());
+    }
+
+    /**
+     * Retrieves a TTS service.
+     * If a default name is configured and the service available, this is returned. Otherwise, the first available
+     * service is returned.
+     *
+     * @return a TTS service or null, if no service is available or if a default is configured, but no according service
+     *         is found
+     */
+    public TTSService getTTS() {
+        TTSService tts = null;
+        if (defaultTTS != null) {
+            tts = ttsServices.get(defaultTTS);
+            if (tts == null) {
+                logger.warn("Default TTS service '{}' not available!", defaultTTS);
+            }
+        } else if (ttsServices.size() > 0) {
```

Comment:
If we are not interested in the real size but check if there is an element, we should use isEmpty() (or the negative result of that). Size could be more expensive (runtime) then isEmpty